### ISSUE TITLE
Search all metadata fields with the text search filter

### ISSUE THREAD

**Issue by PulsarFX** [darktable-org/darktable#13747]
**Is your feature request related to a problem? Please describe.**
Branched from  #13712
See for more context

**Describe the solution you'd like**
Text search filter should look into any metadata field there is for a file.

**Comment by AlicVB** [comment]
I won't have the time to work on this, but if someone want to give it a try, I think the main place to look is : 
https://github.com/darktable-org/darktable/blob/1c6476362220b985ad1355e6d010f3755633857b/src/common/collection.c#L1930-L1939
Keep in mind as the search should stay fast enough in order to make the autocompletion usable...

**Comment by zisoft** [comment]
This line

https://github.com/darktable-org/darktable/blob/1c6476362220b985ad1355e6d010f3755633857b/src/common/collection.c#L1932

already takes into account all metadata fields (creator, description, notes, ...).

@PulsarFX can you please clarify what you are missing?

**Comment by PulsarFX** [comment]
@zisoft please have a look at #13712 in section "To Reproduce", where I've described the search behavior and results including screenshots.

tl;dr: I can't find camera and lens information via the search text filder. It seems as if `main.meta_data` doesn't contain the camera or lens information and another table must be searched?
